mautic/mautic
# Progressive profiling

Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | N
| New feature?  | Y
| BC breaks?    | N
| Deprecations? | N
| Fixed issues  | https://github.com/mautic/mautic/issues/414, https://github.com/mautic/mautic/issues/1242
| Documentation | https://github.com/mautic/documentation/pull/63

## Description
Progressive Profiling allows to display only certain fields in the form for current contact based on the past submissions. The conditions are:

- If the form field value already exists, don't display the field.
- Display the field only if the contact submitted this form for X times.

## Steps to test this PR
Run `app/console doctrine:migrations:migrate` command, clear the cache. Build a form with several fields. Each field has the new __Behavior__ tab where you can configure the conditions described above. Configure those conditions so you'd test as many combinations you can think of. Go to the form preview and check if the form displayed the correct fields. Submit the form and check the fields again. Repeat until you test the highest number of submissions you've configured.

The whole form field and action builder was redesigned, so make sure that everything like sorting, delete, new field/action, update of a field/action works.

Even the preview (and all means of displaying a Mautic form in production) was re-styled a bit. The fields are not always 50% wide, but instead, they are in a container with `min-width: 600px` and this box is centered. I hope it will be more convenient default styling.
